listenbrainz_spark/recommendations/create_dataframes.py

```diff
@@ -0,0 +1,174 @@
+    listens_df = run_query("""
+            SELECT listened_at
+                 , track_name
+                 , recording_msid
+                 , user_name
+             From %s
+        """ % (table))
+    return listens_df
+
+def prepare_recording_data(table):
+    """ Prepare recordings dataframe to select distinct recordings/tracks
+        listened to and assign each recording a unique integer id.
+
+        Args: 
+            table: Registered dataframe to run SQL queries
+
+        Returns:
+            recordings_df: recordings dataframe with columns as:
+                track_name, recording_msid, artist_name, artist_msid, 
+                release_name, release_msid, recording_id
+    """
+    recordings_df = run_query("""
+            SELECT track_name
+                 , recording_msid
+                 , artist_name
+                 , artist_msid
+                 , release_name
+                 , release_msid
+                 , row_number() over (ORDER BY "recording_msid") AS recording_id
+             From (SELECT DISTINCT recording_msid, track_name, artist_name, artist_msid, 
+                    release_name, release_msid FROM %s)
+        """ % (table))
+    return recordings_df
+
+def get_playcounts_data(listens_df, users_df, recordings_df):
+    """ Prepare playcounts dataframe by joining listens_df, users_df,
+        recordings_df to select distinct tracks that a user has listened to
+        for all the users along with listen count.
+
+        Args:
+            listens_df: Listens dataframe
+            users_df: Users dataframe
+            recordings_df: Recordings dataframe
+
+        Returns:
+            playcounts_df: playcounts dataframe with columns as:
+                user_id, recording_id, count
+    """
+    listens_df.createOrReplaceTempView('listen')
+    users_df.createOrReplaceTempView('user')
+    recordings_df.createOrReplaceTempView('recording')
+    playcounts_df = run_query("""
+        SELECT user_id,
+               recording_id,
+               count(recording_id) as count
+          FROM listen
+    INNER JOIN user
+            ON listen.user_name = user.user_name
+    INNER JOIN recording
+            ON recording.recording_msid = listen.recording_msid
+      GROUP BY user_id, recording_id
+      ORDER BY user_id
+    """)
+    return playcounts_df
+
+def main():
+    ti = time.time()
+    try:
+        listenbrainz_spark.init_spark_session('Create_Dataframe')
+    except Exception as err:
+        raise SystemExit("Cannot initialize Spark Session: %s. Aborting..." % (str(err)))
+
+    df = None
+    for y in range(config.STARTING_YEAR, config.ENDING_YEAR + 1):
+        for m in range(config.STARTING_MONTH, config.ENDING_MONTH + 1):
+            try:
+                month = listenbrainz_spark.sql_context.read.parquet('{}/data/listenbrainz/{}/{}.parquet'.format(config.HDFS_CLUSTER_URI, y, m))
+                df = df.union(month) if df else month
+            except Exception as err:
```

Comment:
Having this blanket exceptions is really not ideal. You should try and find exceptions that are appropriate for the given tools you are using. They seem to be poorly documented, but I cloned the spark repo and grepped in the pyspark dir and found:

  https://github.com/apache/spark/blob/master/python/pyspark/sql/utils.py

Reply:
Thanks for the link:)